packages/plugins/AddressSearch/src/store/actions.ts

```diff
@@ -2,217 +2,231 @@ import debounce from 'lodash.debounce'
+      getters.searchMethods
+        .filter(({ url }) => !url)
+        .forEach((miss) =>
+          console.error(
+            `POLAR Plugin AddressSearch: A specification is missing an URL: (${JSON.stringify(
+              miss
+            )})`
+          )
         )
-      )
 
-    // add custom added search implementations, if any configured
-    const customSearchMethods =
-      getters.addressSearchConfiguration.customSearchMethods
-    if (customSearchMethods) registerSearchMethods(customSearchMethods)
-  },
-  setSelectedGroupId(
-    { commit, dispatch, state },
-    selectedGroupId: string
-  ): void {
-    commit('setSelectedGroupId', selectedGroupId)
+      methodContainer = getMethodContainer()
 
-    /* whenever the selected search group name changes,
-     * redo input – if it triggers a search, the user
-     * will probably want to see the results in new
-     * search service group */
-    commit('setSearchResults', SearchResultSymbols.NO_SEARCH)
-    dispatch('input', state.inputValue)
-  },
-  input({ commit, dispatch }, input: string): void {
-    commit('setInputValue', input)
-    dispatch('abortAndRequest')
-  },
-  abortAndRequest() {
-    if (abortController) {
-      abortController.abort()
-      abortController = undefined
-    }
-    debouncedLoad()
-  },
-  clear({ commit }): void {
-    commit('setSearchResults', SearchResultSymbols.NO_SEARCH)
-    commit('setChosenAddress', null)
-  },
-  load({
-    state: { inputValue },
-    rootGetters,
-    getters,
-    commit,
-    dispatch,
-  }): Promise<void> | void {
-    const { minLength } = getters.addressSearchConfiguration
-    const activeSearchMethods = getters.selectedGroup
-    // Value is null when the input is cleared; extra undefined check for safety
-    if (
-      typeof inputValue === 'undefined' ||
-      inputValue === null ||
-      inputValue.length < minLength
-    ) {
-      commit('setSearchResults', SearchResultSymbols.NO_SEARCH)
-      dispatch('indicateLoading', false)
-      return
-    }
-    dispatch('indicateLoading', true)
-    abortController = new AbortController()
-    const localAbortControllerReference = abortController
-    const searchPromises: Promise<FeatureCollection>[] =
-      activeSearchMethods.map((method) =>
-        getSearchMethod(method.type)(
-          abortController.signal,
-          method.url,
-          inputValue,
-          {
-            ...method.queryParameters,
-            epsg: rootGetters.configuration.epsg,
-            map: rootGetters.map,
-          }
+      // add custom added search implementations, if any configured
+      const customSearchMethods =
+        getters.addressSearchConfiguration.customSearchMethods
+      if (customSearchMethods) {
+        methodContainer.registerSearchMethods(
+          Object.fromEntries(
+            Object.entries(customSearchMethods).map(([key, value]) => [
+              key,
+              value.bind(this),
+            ])
+          )
```

Comment:
Why is this extra step necessary now?

Reply:
The self-registered functions did not have the store as `this` context. However, the one from DISH needs access to the store for firing toasts. Hence I've bound all custom search methods to the store for easy access.